angular-map/src/app/map/reverse-geocoding.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@angular/core';
+import { Observer } from 'rxjs/Observer';
+import { Observable } from 'rxjs/Observable';
+import { AgmCoreModule, MapsAPILoader } from '@agm/core';
+
+
+@Injectable() export class ReverseGeocodingService {
+
+    google: any;
+    geocoder: google.maps.Geocoder;
+    place: any;
+
+
+
+    constructor(private mapsAPILoader: MapsAPILoader) {
+        try {
+            this.mapsAPILoader.load().then(() => {
+                this.geocoder = new google.maps.Geocoder();
+            })
```

Comment:
the `this.mapsAPILoader.load().then(() => {
this.geocoder = new google.maps.Geocoder();
})` takes longer than the `ngOnChanges()` in the map component which is asking for a place because the latitude and longitude has changed from nothing to the defaults